### ISSUE TITLE
Memory leak caused by `@tool` scripts can crash the editor

### ISSUE THREAD

**Issue by Yunxi-awa** [godotengine/godot#94948]
### Tested versions

Very likely to appear in all 4.x Godot versions

### System information

Godot v4.3.rc1.mono - Windows 10.0.19045 - Vulkan (Mobile) - Memory 31.8GiB

### Issue description

[I used a translation tool] I mistakenly wrote code that could cause a memory leak. When I was testing the game, it caused the editor to crash. When I tried to restart the project again, the editor crashed again. After a period of investigation, I found that the problem was caused by a file with the `@tool` tag. Delete `@tool` and write code correctly will solve it.

But I would like to make a suggestion, can we ensure that the editor is not killed by the system when memory leak occurs ( Just stop the game related calculations and rendering ) , and remind the user of the memory leak issue to avoid troubleshooting for more than 20 minutes like I did.

### Steps to reproduce

Simply put,
1> Writing can cause memory leaks
2> Add an evil @ tool tag at the beginning of the file
3> Try to have the editor run this file. If you don't remove the @ tool tag, your project will never start

### Minimal reproduction project (MRP)

N/A

**Comment by AThousandShips** [comment]
This is not specific enough, please upload an MRP to make this possible to test:
- A small Godot project which reproduces the issue, with no unnecessary files included. Be sure to not include the `.godot` folder in the archive (but keep `project.godot`).
- Having an MRP is very important for contributors to be able to reproduce the bug in the same way that you are experiencing it. When testing a potential fix for the issue, contributors will use the MRP to validate that the fix is working as intended.
- Drag and drop a ZIP archive to upload it (max 10 MB). **Do not select another field until the project is done uploading.**
- **Note for C# users:** If your issue is *not* C#-specific, please upload a minimal reproduction project written in GDScript. This will make it easier for contributors to reproduce the issue locally as not everyone has a .NET setup available.

**Comment by AThousandShips** [comment]
Closing due to lack of response, if you provide the requested information this can be reopened
